strimzi/tests/test_unit.py

```diff
@@ -101,3 +120,34 @@ def test_instance_without_operator_endpoint(dd_run_check, check, instance):
         "`cluster_operator_endpoint`, `topic_operator_endpoint` or `user_operator_endpoint`.",
     ):
         dd_run_check(check(instance))
+
+
+@pytest.mark.parametrize(
+    "namespace, instance, endpoint_key",
+    [
+        (
+            "cluster_operator",
+            MOCKED_CLUSTER_OPERATOR_INSTANCE,
+            "cluster_operator_endpoint",
+        ),
+        (
+            "topic_operator",
+            MOCKED_TOPIC_OPERATOR_INSTANCE,
+            "topic_operator_endpoint",
+        ),
+        (
+            "user_operator",
+            MOCKED_USER_OPERATOR_INSTANCE,
+            "user_operator_endpoint",
+        ),
+    ],
+)
+def test_parse_config_populates_only_configured_scrapers(
+    namespace, instance, endpoint_key
+):
+    strimzi = StrimziCheck("strimzi", {}, [instance])
+    strimzi.parse_config()
+    assert len(strimzi.scraper_configs) == 1
+    assert strimzi.scraper_configs[0][endpoint_key] == instance[endpoint_key]
+    assert strimzi.scraper_configs[0]["openmetrics_endpoint"] == instance[endpoint_key]
+    assert strimzi.scraper_configs[0]["namespace"] == "strimzi." + namespace
```

Comment:
if you want to go that route and test the different possibilities, you should also test when you have two or three urls provided in the same instance: 

```
{
    "cluster_operator_endpoint": f"foo",
    "topic_operator_endpoint": f"bar",
    "user_operator_endpoint": f"baz",
}
```

This should generate 3 scraper_configs